File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/clr_fundamentals/samples/typemembers/events/vb/Events.vb
Imports System


Class Button
   ' Declare a Click Event allowing other objects to register themselves with this event
   Public Event Click As EventHandler

   Public Sub SimulateClick()
      ' Fire the event         
      RaiseEvent Click(Me, Nothing)
   End Sub
End Class


Class App
   ' This method is called when the Button object is clicked
   Sub OnClick(sender As Object, args As EventArgs)
      Console.WriteLine("App object received the Click notification from the Button object.")
   End Sub

   Shared Sub Main()
      ' Create a button object
      Dim btn As New Button()

      ' Create an instance of our App class
      Dim app As New App()

      ' Register a callback method with the Button object's Click event
      AddHandler btn.Click, AddressOf app.OnClick

      ' Simulate that the button has been clicked
      btn.SimulateClick()

      Console.Write("Press Enter to close window...")
      Console.Read()
   End Sub
End Class

File: vs2003_cd01/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/Samples/Tutorials/clr_fundamentals/samples/typemembers/events/cs/Events.cs
using System;


class Button {
   // Declare a Click Event allowing other objects to register themselves with this event
   public event EventHandler Click;

   public void SimulateClick() {
      // If Click is null, then no objects have registered themselves with the event
      if (Click != null) {

         // At least one object is registered with the event, fire the event         
         Click(this, null);
      }
   }
}


class App {
   // This method is called when the Button object is clicked
   void OnClick(Object sender, EventArgs args) {
      Console.WriteLine("App object received the Click notification from the Button object.");
   }

   static void Main() {
      // Create a button object
      Button btn = new Button();

      // Create an instance of our App class
      App app = new App();

      // Register a callback method with the Button object's Click event
      btn.Click += new EventHandler(app.OnClick);

      // Simulate that the button has been clicked
      btn.SimulateClick();

      Console.Write("Press Enter to close window...");
      Console.Read();
   }
}
